openstack_plugin_common/__init__.py

```diff
@@ -978,11 +996,169 @@ def wrapper(*args, **kw):
             else:
                 raise
         except keystone_exceptions.ClientException as e:
+            _check_valid_resource_id_with_operation(kw, True)
             _re_raise(e, recoverable=False)
 
     return wrapper
 
 
+def handle_conf_cycle_operations(_ctx, operation_name, instance_id, exception):
+    """
+    function to handle CLOUDIFY_CONF_CYCLE_OPERATION given exception
+    if no exception the function add runtime property for that operation
+    if exception remove the runtime property, handle start/stop operations flag
+    :param _ctx:
+    :param operation_name:
+    :param instance_id:
+    :param exception:
+    :return:
+    """
+    op_prefix = operation_name.split(".")[-1]
+    runtime_prop = op_prefix + '_' + instance_id
+
+    if not exception:
+        # if runtime for this operation already set return false
+        if _ctx.instance.runtime_properties.get(runtime_prop):
+            _ctx.logger.info(op_prefix + ' already done')
+            return False
+        else:
+            # set runtime property of the operation and return true
+            _ctx.instance.runtime_properties[runtime_prop] = True
+
+            # if operation start , delete stop property if exists
+            if operation_name == CLOUDIFY_START_OPERATION and \
+                    _ctx.instance.runtime_properties.get(
+                        'stop_' + instance_id):
+                del _ctx.instance.runtime_properties['stop_' +
```

Comment:
Need to check that such flag really exist before delete, or on final step of delete action clean up whole runtime properties instead delete flags one by one.

Reply:
these just to delete stop , in case of start [ negate the reverse action flag ]